server/src/main/java/org/opensearch/bootstrap/BootstrapChecks.java

```diff
@@ -720,10 +722,10 @@ public final BootstrapCheckResult check(BootstrapContext context) {
 
         @SuppressWarnings("removal")
         boolean isAllPermissionGranted() {
-            final SecurityManager sm = System.getSecurityManager();
-            assert sm != null;
+            final Policy policy = AgentPolicy.getPolicy();
+            assert policy != null;
             try {
-                sm.checkPermission(new AllPermission());
+                AgentPolicy.checkPermission(new AllPermission());
```

Comment:
Do we support this?

Reply:
> Do we support this?

Yes, for sure - this checks just makes sure no policy has `AllPermission`